File: src/SemanticTypeForgers/forger.tests.jl
include("forger.jl")
using .Forger

behavior = AtomicBehavior{Any}("test")

@assert Forger.getPrimitiveType(behavior, 5) == 5
@assert Forger.convertToPrimitive(behavior, " 123 ") == 123
@assert Forger.convertToPrimitive(behavior, "true") == true
@assert Forger.convertToPrimitive(behavior, "{}") == 0

val = Forger.proccessValue(behavior, Dict(
    "productPrice" => "10.5",
    "deliveryPrice" => 2.5
))

@assert val == 13.0 "Test failed: expected 13.0 got $val"

emailBehavior = AtomicBehavior{Any}("PersonEmail")
@assert Forger.validate(emailBehavior, "test@example.com") == true
@assert Forger.validate(emailBehavior, "invalid-email") == false

try
    Forger.forge(emailBehavior, "invalid-email")
    error("Test failed: should have thrown")
catch e
    @assert occursin("Validation failed", e.msg)
end

println("Julia tests passed")

File: src/SemanticTypeForgers/forger.tests.py
import sys
from forger import AtomicBehavior

def test():
    behavior = AtomicBehavior("test")

    assert behavior["getPrimitiveType"](5) == 5
    assert behavior["convertToPrimitive"](" 123 ") == 123
    assert behavior["convertToPrimitive"]("true") is True

    # Empty object translates to 0
    assert behavior["convertToPrimitive"]({}) == 0

    val = behavior["proccessValue"]({"productPrice": "10.5", "deliveryPrice": 2.5})
    assert val == 13.0, f"Expected 13.0, got {val}"

    emailBehavior = AtomicBehavior("PersonEmail")
    assert emailBehavior["validate"]("test@example.com") is True
    assert emailBehavior["validate"]("invalid-email") is False

    try:
        emailBehavior["forge"]("invalid-email")
        assert False, "Should have thrown an exception"
    except Exception as e:
        assert "Validation failed" in str(e)

    print("Python tests passed")

if __name__ == "__main__":
    test()
